controllers/proxy/ecr_proxy.go

```diff
@@ -15,9 +16,18 @@ func NewECRProxy(registryEndpoint string) ProxyFunc {
 	})
 
 	return ProxyFunc(func(token string, w http.ResponseWriter, r *http.Request) {
+		originalHost := r.Host
 		r.Header.Set("Authorization", fmt.Sprintf("Basic %s", token))
 		r.Host = registryEndpoint
 
+		reverseProxy.ModifyResponse = func(resp *http.Response) error {
```

Comment:
Damn, this is a nasty one. 

Did you end up stumbling on to this by digging into how nginx does proxying? Because this fix directly links back to that nginx config block that you suspected from the get-go:
https://en.wikipedia.org/wiki/HTTP_location

In other words, the permanent 301 we had setup for the `/v2` endpoint was, by default, re-writing the `Location` header. 

Nginx docs for reference: http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_redirect